interfaces/apparmor/backend_test.go

```diff
@@ -2498,6 +2498,7 @@ func (s *backendSuite) TestRemoveAllSnapAppArmorProfiles(c *C) {
 		c.Check(os.IsNotExist(err), Equals, true)
 	}
 
+	// remove snaps to restore the test environment back to the original state
```

Comment:
Thanks for this comment! If this is about cleanup maybe a nicer way would be to do something like: 
```go
diff --git a/interfaces/apparmor/backend_test.go b/interfaces/apparmor/backend_test.go
index 8edc020771..94b2b3abcd 100644
--- a/interfaces/apparmor/backend_test.go
+++ b/interfaces/apparmor/backend_test.go
@@ -2479,7 +2479,9 @@ func (s *backendSuite) TestRemoveAllSnapAppArmorProfiles(c *C) {
 
        opts := interfaces.ConfinementOptions{}
        snapInfo1 := s.InstallSnap(c, opts, "", ifacetest.SambaYamlV1, 1)
+       defer s.RemoveSnap(c, snapInfo1)
        snapInfo2 := s.InstallSnap(c, opts, "", ifacetest.SomeSnapYamlV1, 1)
+       defer s.RemoveSnap(c, snapInfo2)
        snap1nsProfile := filepath.Join(dirs.SnapAppArmorDir, "snap-update-ns.samba")
        snap1AAprofile := filepath.Join(dirs.SnapAppArmorDir, "snap.samba.smbd")
        snap2nsProfile := filepath.Join(dirs.SnapAppArmorDir, "snap-update-ns.some-snap")
@@ -2497,8 +2499,4 @@ func (s *backendSuite) TestRemoveAllSnapAppArmorProfiles(c *C) {
                _, err := os.Stat(p)
                c.Check(os.IsNotExist(err), Equals, true)
        }
-
-       // remove snaps to restore the test environment back to the original state
-       s.RemoveSnap(c, snapInfo2)
-       s.RemoveSnap(c, snapInfo1)
 }
````
(or s.AddCleanup() instad of defer but the idea is the same)

Reply:
Ah that is definitely nicer - added in https://github.com/snapcore/snapd/pull/12768/commits/090cd3f1e153a9b083459b8050ac77a680464243